src/solve/optim.jl

```diff
@@ -61,7 +61,16 @@ function __solve(prob::OptimizationProblem, opt::Optim.AbstractOptimizer,
     maxiters = _check_and_convert_maxiters(maxiters)
     maxtime = _check_and_convert_maxtime(maxtime)
 
-    f = instantiate_function(prob.f,prob.u0,prob.f.adtype,prob.p)
+    if prob.sense === MaxSense
+        obj = (args...) -> -prob.f.f(args...)
+        kwargs_tup = (grad = prob.f.grad === nothing ? nothing : (args...) -> -prob.f.grad(args...),
+                      hess = prob.f.hess === nothing ? nothing : (args...) -> -prob.f.hess(args...),
+                      hv = prob.f.hv === nothing ? nothing : (args...) -> -prob.f.hv(args...), )
+        optfun = OptimizationFunction(obj, prob.f.adtype; kwargs_tup...)
```

Comment:
This should probably never be done internally.

Reply:
Handle it here: https://github.com/SciML/GalacticOptim.jl/pull/170/files#diff-60eefb6ae5e73d80fe740490f1d994b59a3e7dea90a11c624effd5e569fe981fR78-R84

Reply:
The link doesn't seem to point me to anything.. but other than that what do you mean, creating a new `OptimizationFunction` shouldn't be done there?